### DISCUSSION TITLE
Issue with Keycloak Deployment Using IIS Reverse Proxy: Port Parsing Failure

### DISCUSSION THREAD

**Discussion by buzzo123** [keycloak/keycloak#24800] [Q&A]
### Description:
**Environment:**
- Keycloak version: 22.0.5
- IIS version: 10.0.19041.1
- Deployment type: Standalone

**Issue:**
I am encountering a problem while deploying Keycloak behind an IIS reverse proxy. Despite correctly setting up the necessary environment variables for reverse proxying (`X-Forwarded-For`, `X-Forwarded-Proto`, `X-Forwarded-Host`, and `X-Forwarded-Port`), Keycloak fails to parse the forwarded port and defaults to -1.

**Error Message:**
```
ERROR [io.quarkus.vertx.http.runtime.ForwardedParser] (executor-thread-7) Failed to parse a port from "forwarded"-type headers, using the default port -1
```

The logs confirm that the `X-Forwarded-Port` is correctly set to 443, yet Keycloak does not seem to recognize this. I've attached the relevant log snippet for reference.

**Attachment:**
![image](https://github.com/keycloak/keycloak/assets/2952183/3f3b1272-3ffd-4b70-84a0-0142d3d88e32)

**Configuration Details:**
this is my web.config:
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <rewrite>
      <rules>
        <clear />        
        <rule name="Reverse Proxy to Auth" enabled="true" stopProcessing="true">
          <match url="(.*)" />
          <conditions logicalGrouping="MatchAll" trackAllCaptures="false" />
          <action type="Rewrite" url="http://localhost:8080/auth/{R:1}" logRewrittenUrl="true" />
          <serverVariables>
            <set name="HTTP_X_FORWARDED_PROTO" value="https" />
            <set name="HTTP_X_FORWARDED_PORT" value="443" />
            <set name="HTTP_X_FORWARDED_HOST" value="https://mydomain/" />
          </serverVariables>
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```

this my keycloak.conf file:
```
db=mssql
transaction-xa-enabled=false
db-username=user
db-password=Passw0rd
db-url=jdbc:sqlserver://localhost:1440;databaseName=auth;encrypt=false
proxy=edge

hostname-url=https://mydomain/auth/
hostname-admin-url=https://mydomain/auth/
http-relative-path=/auth
```
**Steps to Reproduce:**
1. bin/kc.bat build
2. bin/kc.bat start

**Expected Behavior:**
Keycloak should correctly parse the port from the `X-Forwarded-Port` header and not default to -1.

**Actual Behavior:**
Keycloak fails to parse the port from the `X-Forwarded-Port` header, defaulting to -1.

I would appreciate any guidance or assistance in resolving this issue. Thank you!

**Comment by esekamachi** [comment]
Hello, have you fix this problem?

**Reply by devshrm** [reply]
proxy should be `xforwarded`

https://www.keycloak.org/server/hostname
